x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/optimizer/OptimizerRunTests.java

```diff
@@ -48,4 +57,57 @@ public void testWhereClause() {
         LogicalPlan p = plan("SELECT some.string l FROM test WHERE int IS NOT NULL AND int < 10005 ORDER BY int");
         assertNotNull(p);
     }
-}
\ No newline at end of file
+
+    public void testReducedBinaryComparisonGreaterThenOrEqual() {
```

Comment:
I would add some tests for testing with numbers where overflow or precision loss can occur.

Reply:
I've added more tests.